DataFormats/Detectors/TRD/include/DataFormatsTRD/HelperMethods.h

```diff
@@ -0,0 +1,74 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_TRD_HELPERMETHODS_HH
+#define ALICEO2_TRD_HELPERMETHODS_HH
+
+#include "DataFormatsTRD/Constants.h"
+
+namespace o2
+{
+namespace trd
+{
+
+struct HelperMethods {
+  static int getROBfromPad(int irow, int icol)
+  {
+    return (irow / constants::NMCMROBINROW) * 2 + getColSide(icol);
+  }
+
+  static int getMCMfromPad(int irow, int icol)
+  {
+    if (irow < 0 || icol < 0 || irow > constants::NROWC1 || icol > constants::NCOLUMN) {
+      return -1;
```

Comment:
I know this is written simply analog to what was there in AliRoot, but now that I see it again and think about it shouldn't we put an error message here in case of a wrong input? Because where these functions are used the return values are not checked for plausibility.

Reply:
These methods can be called a lot. Since the return value is a flag for errors, I don't think one should add messages here, even if it's just a DEBUG message. So, I would leave this as it is or add an assertion for the whole test condition.

Reply:
Especially if the function is called a lot, we should probably not force the user to check for errors. I think this should be a fatal error: if we do not get irow and icol correct, then there is a major issue somewhere else in the code.

Reply:
My point here is that this should never happen. These methods are primarily internal, and if there is an error, it should pop up in a test (less likely, be caught eventually as a bug when you start seeing -1 around). Therefore it will require a change in the codebase. So, we expect that this actually never happens or rarely happens. That's why we should use assert, instead of logging, IMO.

We had a similar discussion with @sawenzel a long time ago (See: [this code review](https://github.com/AliceO2Group/AliceO2/pull/3389#discussion_r416429688)), and the choice was using assert.

Since this condition should never happen or at least is very unlikely, one could use some optimizations, like: https://en.cppreference.com/w/cpp/language/attributes/likely, but I am not sure how this applies to O2. Of course, this is not relevant if one drops the conditional here.